### ISSUE TITLE
Add #ENCODING:UTF8 tag in txt file to languages with special characters

### ISSUE THREAD

**Issue by stripu** [rakuri255/UltraSinger#65]
Great work implementing the alternate language models, it works great, and I converted a romanian song pretty nicely with UltraSinger.
The only problem i had when using it was that special characters are being displayed improperly. For these situations, UltraStar/Vocaluxe implemented a special tag that has to be added to the files:
#ENCODING:UTF8
So if this could be added automatically to languages that use special characters (German, Romanian, Czech, etc) that would be awesome. 

Thanks again, and I'll contribute with feedback on other matters :)

**Comment by bohning** [comment]
It is only Vocaluxe that either needs this tag (or UTF8 with BOM) to determine the file encoding successfully. USDX detects UTF8 just fine without needing either, so since Vocaluxe is not actively developed anymore, I suggest that adding #ENCODING to the file should at least be optional (with the default not adding it to the text file).

There is a test build where Vocaluxe assumes UTF8 by default and only uses different encodings (e.g. CP1252) when it is stated in the #ENCODING tag. I think this should be the way to go. Besides USDX, UltraStar Manager, YASS and Performous all handle UTF8 just fine afaik.

**Comment by stripu** [comment]
Indeed, I tested the output files with UltraStar and it works okay. 
I use the nightly build of Vocaluxe (i just find it cooler than UltraStar, as it also has the web server and a bit better interface) and without the tag, fonts are looking bad.
Indeed, a solution would be to add this as optional in the Settings.py file

**Comment by bohning** [comment]
It’s rather a workaround than a solution. A real solution would be to update Vocaluxe to automatically detect the file encoding without relying on either BOM or #ENCODING tag.

As an intermediate step, @flokuep created a test build that defaults to UTF8 if there is neither BOM nor #ENCODING (instead of CP1252), but the downside is that existing CP1252 files with special characters would be displayed as UTF8 and would therefore have issues. The solution to this would be easy: everybody using Vocaluxe would then just have to convert existing CP1252 files to UTF8 (e.g. using UltraStar Manager, which *always* outputs UTF8 in order to finally get rid of anything other than UTF8). There is no need for any other encoding other than UTF8 if every software properly supported it.
